File: haskell/src/TestLibrary.hs
module TestLibrary (modifyArray, invertStringCase, exponentiate, identity, chooseLeft, reverseList, applyFunction) where

import Data.Char
import Data.Int

modifyArray :: [Int32] -> [Int32]
modifyArray = zipWith (*) [1..]

invertStringCase :: [Char] -> [Char]
invertStringCase = fmap invertCase

exponentiate :: Int32 -> Int32 -> Int32
exponentiate x y = x^y

identity :: a -> a
identity x = x

chooseLeft :: a -> b -> a
chooseLeft x _ = x

reverseList :: [a] -> [a]
reverseList [] = []
reverseList (x : xs) = reverseList xs ++ [x]

applyFunction :: (Int32 -> Int32) -> Int32 -> Int32
applyFunction f = f

invertCase :: Char -> Char
invertCase c | isLower c = toUpper c
             | isUpper c = toLower c
             | otherwise = c

File: python/lib/test-library.py
def library_language():
    return [112,121,116,104,111,110]

def binary_or(x, y):
    return x or y

def modify_array(x):
    for i in range(0, len(x)):
        x[i] = x[i] * (i + 1)
    return x

def exponentiate(x, y):
    return x ** y

def identity(x):
    return x

def choose_left(x, y):
    return x

def reverse_list(l):
    return list(reversed(l))

def apply_function(f, x):
    return f(x)
